public/components/model_version/__tests__/model_version.test.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { generatePath, Route } from 'react-router-dom';
+import userEvent from '@testing-library/user-event';
+
+import { ModelVersion } from '../model_version';
+import { render, screen, waitFor, mockOffsetMethods } from '../../../../test/test_utils';
+import { routerPaths } from '../../../../common/router_paths';
+
+const setup = () =>
+  render(
+    <Route path={routerPaths.modelVersion}>
+      <ModelVersion />
+    </Route>,
+    { route: generatePath(routerPaths.modelVersion, { id: '1' }) }
+  );
+
+describe('<ModelVersion />', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should display consistent model name and version', async () => {
+    setup();
+
+    await waitFor(() => {
+      expect(screen.getByText('model1')).toBeInTheDocument();
+      expect(screen.getByText('v1.0.0')).toBeInTheDocument();
+    });
+  });
+
+  it('should display loading screen during Model.getOne calling', async () => {
+    jest.useFakeTimers();
+
+    setup();
+
+    expect(screen.getByTestId('modelVersionLoadingSpinner')).toBeInTheDocument();
+    jest.advanceTimersByTime(3000);
```

Comment:
Q: where is this 3000 from?

Reply:
I since I removed some Model.prototype.getOne API mock here. Will update the code here.